Modules/Loadable/Markups/MRML/vtkSlicerDijkstraGraphGeodesicPath.h

```diff
@@ -0,0 +1,87 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, Cancer
+  Care Ontario, OpenAnatomy, and Brigham and Women�s Hospital through NIH grant R01MH112748.
+
+==============================================================================*/
+
+#ifndef __vtkSlicerDijkstraGraphGeodesicPath_h
+#define __vtkSlicerDijkstraGraphGeodesicPath_h
+
+// VTK includes
+#include <vtkDijkstraGraphGeodesicPath.h>
+
+// export
+#include "vtkSlicerMarkupsModuleMRMLExport.h"
+
+//
+// Set built-in type.  Creates member Set"name"() (e.g., SetVisibility());
+//
+#define vtkSetAdjacencyMacro(name,type) \
+virtual void Set##name (type _arg) \
+{ \
+  vtkDebugMacro(<< this->GetClassName() << " (" << this << "): setting " #name " to " << _arg); \
+  if (this->name != _arg) \
+  { \
+    this->RecalculateAdjacency = true;\
+    this->name = _arg; \
+    this->Modified(); \
+  } \
+}
+
+/// Filter that generates curves between points of an input polydata
+class VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkSlicerDijkstraGraphGeodesicPath : public vtkDijkstraGraphGeodesicPath
+{
+public:
+  vtkTypeMacro(vtkSlicerDijkstraGraphGeodesicPath, vtkDijkstraGraphGeodesicPath);
+  static vtkSlicerDijkstraGraphGeodesicPath* New();
+  void PrintSelf(ostream& os, vtkIndent indent) override;
+
+  enum
+  {
+    COST_FUNCTION_DISTANCE,
+    COST_FUNCTION_ADDITIVE,
```

Comment:
COST_FUNCTION_SCALING (or COST_FUNCTION_SCALE, or COST_FUNCTION_MULTIPLICATIVE) would be a very basic type, too, it would be nice to be able to experiment with it